### PR TITLE
Website 78 replace moment dependency 2

### PR DESCRIPTION
**Pull request by SalazarJosh**
# Overview
Moment.js is no longer supported and was built on an old era of the JS ecosystem. Replacement options included date-fns _[[website](https://date-fns.org/), [github](https://github.com/date-fns/date-fns)]_ and day.js _[[website](https://day.js.org/), [github](https://github.com/iamkun/dayjs)]_. I initially decided [to go with date-fns](https://github.com/mlibrary/lib.umich.edu/pull/252) since it appears to have a larger user base and more support, but the amount of rewriting required to make date-fns work including the different architecture of Date() objects as opposed to moment objects made using date-fns difficult.

After having repeat build errors with date-fns, I was not confident that it would be a good moment replacement. I took another look at dayjs and found it to be a very good candidate for replacing moment. Unlike date-fns, most of the functions were the same between moment and dayjs and all the formatting of the date/time strings was the same.

Check out this article to learn a bit more about the differences between moment, dayjs, and date-fns: [You don't (may not) need Moment.js](https://github.com/you-dont-need/You-Dont-Need-Momentjs). I don't think we're losing anything by switching to dayjs over date-fns.

This PR should move everything from moment to dayjs.

All changed components:
- src/components/panels/events-and-exhibits-panel.js
- src/components/panels/featured-and-latest-news.js
- src/components/panels/hours-lite-panel.js
- src/components/panels/hours-panel.js
- src/components/todays-hours.js
- src/templates/news-landing.js
- src/templates/event.js
- src/templates/news.js
- src/utils/events.js
- src/utils/hours.js

[dayjs documentation](https://day.js.org/docs/en/installation/installation)

> This pull request resolves/closes/fixes [WEBSITE-78](https://mlit.atlassian.net/browse/WEBSITE-78).

Let's remove moment with the June website updates.

## Anything else?
[All conversion tracking was done in this document.](https://docs.google.com/spreadsheets/d/1F9PTjNtmXse1CDv4SM8_KAIWGRv4ezkeD-7yNa8D8MQ/edit#gid=289530058) I linked the second tab (Pre Deploy Testing Status (Phase 2)) where I share some additional notes about the changes and the live site with an example of where that component appears. The first tab is my development tracking document for myself.

## Testing
List instructions on how to test the pull request. Some examples:

- Make sure the PR is consistent in these browsers:
  - [x] Chrome
  - [x] Firefox
  - [ ] Safari
  - [x] Edge (the assignee was not able to test the pull request in this browser)
- Run accessibility tests:
  - [ ] WAVE
  - [ ] ARC Toolkit
  - [ ] axe DevTools

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
         "brace-style": ["error", "1tbs"], // Requires one true brace style
         "no-empty-function": "error", // Require an empty function to at least have a comment explaining why
         "no-var": "error", // Discourages using `var` and recommends using `let` or `const` instead,
+        "react/prop-types": 0, // Ignores proptype declaration and validation
```
Comment (erinesullivan) [inline .eslintrc.json:27]: I know how to resolve the errors to this rule. Let me know if you would like help with that.

File: src/components/panels/hours-lite-panel.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
 import Link from '../link';
 import icons from '../../reusable/icons';
 import MEDIA_QUERIES from '../../reusable/media-queries.js';
 import { displayHours } from '../../utils/hours';
 
-export default function HoursLitePanel({ data }) {
+export default function HoursLitePanel ({ data }) {
   const [initialized, setInitialized] = useState(false);
-  const { field_title } = data;
+  const { field_title: fieldTitle } = data;
```
Comment (erinesullivan) [inline src/components/panels/hours-lite-panel.js:13]: Clever approach!

File: src/components/panels/hours-panel.js [hunk 3]
```diff
@@ -87,72 +87,74 @@ export function HoursPanelNextPrev({ location }) {
...
-          ) : (
-            <Icon icon="navigate_next" />
-          )}
+          {type === 'previous'
+            ? (
+              <Icon icon='navigate_before' />
+              )
+            : (
+              <Icon icon='navigate_next' />
+              )}
```
Comment (erinesullivan) [inline src/components/panels/hours-panel.js:147]: You could simplify this to just one line:
```suggestion
          <Icon icon={ type === 'previous' ? 'navigate_before' : 'navigate_next' } />
```

File: src/templates/event.js [hunk 4]
```diff
@@ -171,33 +172,33 @@ export default function EventTemplate({ data }) {
...
-          to: data.field_registration_link.uri,
+          to: data.field_registration_link.uri
         }
       : null;
   const series = data.relationships.field_event_series?.name;
   const when = dates.map((date) => {
-    const start = moment(date.value);
-    const end = moment(date.end_value);
+    const start = new Date(date.value);
+    const end = new Date(date.end_value);
```
Comment (erinesullivan) [inline src/templates/event.js:188]: Great job on reducing dependency usage!

**Review by erinesullivan — CHANGES_REQUESTED** [review]
Just like the other PR, I did not see anything different from the live site. 🎉 

The biggest change is accidentally importing `date-fns` in `src/templates/news-landing.js`. You can actually avoid using a dependency entirely by using JavaScript's `toLocaleString` method. I suggested the code that would make it work. You also were able to avoid using `dayjs` in `src/templates/event.js`. Have you tried using the vanilla JavaScript approach for the other files so we wouldn't need to use a dependency to replace `moment`?

Using `dayjs` has a couple of advantages over `date-fns`. It seems to have a more seamless approach in replacing `moment`, and it also has a smaller file size. I'm glad you're choosing this dependency instead.

And finally, no ESLint errors and warnings in the changed files. Great job! 👏🏻

File: src/components/panels/hours-lite-panel.js (hunk 2 above)
Comment (SalazarJosh) [reply src/components/panels/hours-lite-panel.js:13]: Thanks! It feels a bit hacky, though. I personally like differentiating between JS variables / imported data variables with snake case and camel case. Have you encountered this yourself? If so, did you have another solution that I could implement for consistency?

File: src/components/panels/hours-panel.js (hunk 3 above)
Comment (SalazarJosh) [reply src/components/panels/hours-panel.js]: Much cleaner! Thanks!

**Comment by SalazarJosh** [comment]
@erinesullivan I've managed to remove dayjs and use vanilla JS throughout. The  only thing I noticed as different is the space between "AM/PM" and the time for events.

File: src/utils/events.js
Comment (SalazarJosh) [inline src/utils/events.js:1]: I'm building out a bunch of strings by pulling the values of the Date object and creating my own strings. I think it's cleaner than having several toLocaleString() / toLocaleDateString() / toLocaleTimeString(). There might be a cleaner way.

**Review by erinesullivan — APPROVED** [review]
No linting errors. Code is solid. Didn't see anything different between production and development. As long as `moment` and `dayjs` get uninstalled in the next round of updates, this is ready to go! 👍🏻

**Comment by SalazarJosh** [comment]
Thanks again for your review(s), Erin! I'll merge it now so I can start with the monthly update and removing moment / date-fns / dayjs!

**Pull request merged**